test/test_check.sh

```diff
@@ -74,7 +74,40 @@ test_it_skips_ignored_paths() {
     {\"ref\": $(echo $ref5 | jq -R .)}
   ]" | jq ".")
   assertEquals "$expected5" "$(check_uri_from_ignoring $repo $ref1 file-c file-b | jq '.')"
+}
+
+test_it_checks_correct_branch() {
+  local repo=$(init_repo)
+  local ref1=$(make_commit $repo)
+  local ref2=$(make_commit_to_branch $repo some-branch )
+
+  # Test initial clone and that check doesn't see descendants that aren't in the given branch
+  local expected1=$(echo "[{\"ref\": $(echo $ref1 | jq -R .)}]"|jq ".")
+  assertEquals "$expected1" "$(check_uri $repo | jq '.')"
+
+  # Test check's first pull is still returning the same result, as above
+  local expected2=$(echo "[{\"ref\": $(echo $ref1 | jq -R .)}]"|jq ".")
+  assertEquals "$expected2" "$(check_uri $repo | jq '.')"
+
+  # Test check doesn't see changes to in other descendant branches when checking from a revision
+  local ref3=$(make_commit_to_branch $repo some-branch )
```

Comment:
Right. Remove this line, but keep line [95][1] and [96][2]?

[1]: https://github.com/Fydon/hg-resource/blob/f56e39c02438610bc13c9c33183486f4f5b38c95/test/test_check.sh#L95
[2]: https://github.com/Fydon/hg-resource/blob/f56e39c02438610bc13c9c33183486f4f5b38c95/test/test_check.sh#L96